docker/test_env/ethereum_env.go

```diff
@@ -0,0 +1,442 @@
+package test_env
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink-testing-framework/docker"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+)
+
+type ConsensusType string
+
+const (
+	ConsensusType_PoS ConsensusType = "pos"
+	ConsensusType_PoW ConsensusType = "pow"
+)
+
+type ExecutionLayer string
+
+const (
+	ExecutionLayer_Geth ExecutionLayer = "geth"
+)
+
+type ConsensusLayer string
+
+const (
+	ConsensusLayer_Prysm ConsensusLayer = "prysm"
+)
+
+type EthereumNetworkBuilder struct {
+	t                 *testing.T
+	l                 zerolog.Logger
+	dockerNetworks    []string
+	consensusType     *ConsensusType
+	consensusLayer    *ConsensusLayer
+	consensusNodes    int
+	executionLayer    *ExecutionLayer
+	executionNodes    int
+	beaconChainConfig *BeaconChainConfig
+	validated         bool
+	existingConfig    *EthereumNetworkConfig
+}
+
+type Eth2Components struct {
+	Geth        *Geth2
+	BeaconChain *PrysmBeaconChain
+	Validator   *PrysmValidator
+}
+
+func NewEthereumNetworkBuilder(t *testing.T) EthereumNetworkBuilder {
+	return EthereumNetworkBuilder{
+		t:              t,
+		l:              logging.GetTestLogger(t),
+		dockerNetworks: []string{},
+		executionNodes: 1,
+		consensusNodes: 1,
+	}
+}
+
+func (b *EthereumNetworkBuilder) WithConsensusType(consensusType ConsensusType) *EthereumNetworkBuilder {
+	b.consensusType = &consensusType
+	return b
+}
+
+func (b *EthereumNetworkBuilder) WithConsensusLayer(consensusLayer ConsensusLayer) *EthereumNetworkBuilder {
+	b.consensusLayer = &consensusLayer
+	return b
+}
+
+func (b *EthereumNetworkBuilder) WithExecutionLayer(executionLayer ExecutionLayer) *EthereumNetworkBuilder {
+	b.executionLayer = &executionLayer
+	return b
+}
+
+func (b *EthereumNetworkBuilder) WithConsensusNodes(consensusNodes int) *EthereumNetworkBuilder {
+	b.consensusNodes = consensusNodes
+	return b
+}
+
+func (b *EthereumNetworkBuilder) WithExecutionNodes(executionNodes int) *EthereumNetworkBuilder {
+	b.executionNodes = executionNodes
+	return b
+}
+
+func (b *EthereumNetworkBuilder) WithBeaconChainConfig(config BeaconChainConfig) *EthereumNetworkBuilder {
+	b.beaconChainConfig = &config
+	return b
+}
+
+func (b *EthereumNetworkBuilder) UsingDockerNetworks(networks []string) *EthereumNetworkBuilder {
```

Comment:
@Tofel any reason to use `Using` instead of `With` here? Just trying to understand the difference :)